### ISSUE TITLE
[EOS 1000D] Preview does not work in auto mode (PTP Operation Not Supported (0x2005))

### ISSUE THREAD

**Issue by larsf96** [gphoto/gphoto2#283]
**Describe the bug**
I've tried to capture a preview image with --capture-preview on my EOS1000D. For some reasons, this does not work when the camera is in automatic mode. When switching the camera to Tv, Av or P modus, i can get preview images without any problem.
When I then switch back to the automatic mode, the first caputure of --capture-preview works as excpected, but the second try fails. I've attached some log files.
The following errors are always showing up for these attempts:
1. PTP Operation Not Supported (0x2005)
2.PTP Device Busy (0x2019)

Is the preview supposed to work in automatic mode?
[notworking.txt](https://github.com/gphoto/gphoto2/files/4066946/notworking.txt)
[working_first_automatic_mode.txt](https://github.com/gphoto/gphoto2/files/4066947/working_first_automatic_mode.txt)
[working_manual_mode.txt](https://github.com/gphoto/gphoto2/files/4066948/working_manual_mode.txt)

**Name the camera**
Canon EOS 1000D 

**libgphoto2 and gphoto2 version**
This version of gphoto2 is using the following software versions and options:
gphoto2         2.5.23.1       gcc, popt(m), exif, no cdk, no aa, no jpeg, no readline
libgphoto2      2.5.22         all camlibs, gcc, ltdl, EXIF
libgphoto2_port 0.12.0         iolibs: disk ptpip serial usb1 usbdiskdirect usbscsi, gcc, ltdl, USB, serial without locking

**To Reproduce**

If its a camera misbehavior, attach debug output using  --debug --debug-logfile=debug.log   (if considered useful).

**Comment by larsf96** [comment]
Any ideas if this can be fixed somehow?

**Comment by msmeissn** [comment]
The PTP Operation Not Supported (0x2005) related to "AfCancel" ... there was a check missing, which is fixed in libgphoto2 2.5.24 now.

For the other its not so clear. need to rty with my EOS 1000D

**Comment by msmeissn** [comment]
I tried with my EOS 1000D with more current libgphoto2 -> no problems to get liveview in either mode.

Please update to 2.5.24. ;)

**Comment by larsf96** [comment]
I've upgraded libgphoto to 2.5.24. The PTP Operation Not Supported (0x2005) error seems to be fixed, but now I get a "Unidentifiable Error".
I've attached the logs
[debug.txt](https://github.com/gphoto/gphoto2/files/4669017/debug.txt)

**Comment by msmeissn** [comment]
I tried back and forth now with my EOS 1000D, but no luck.

Also the camera itself does not go into liveview mode when I press the "Set" button in the automatic modes.

I now looked at the manual of the EOS1000D 

https://www.canon-europe.com/support/consumer_products/products/cameras/digital_slr/eos_1000d.html?type=manuals&language=en

In section 5 "Live View" it indicates this is only possible in the P-TV-AV-M-Adep modes.

So this is a camera limitation, and we cannot do much about that.
